src/Host.cpp

```diff
@@ -802,15 +802,14 @@ bool Host::installPackage(const QString& fileName, int module )
                 pLabel->setText( tr( "Unpacking package:\n\"%1\"\nplease wait..." ).arg( packageName ) );
             }
         }
+        pUnzipDialog->hide(); // Must hide to change WindowModality
         pUnzipDialog->setWindowTitle( tr( "Unpacking" ) );
-        pUnzipDialog->hide();
         pUnzipDialog->setWindowModality( Qt::ApplicationModal );
         pUnzipDialog->show();
         qApp->processEvents();
         pUnzipDialog->raise();
-        pUnzipDialog->show(); // Must do this to ensure modality is applied
-        pUnzipDialog->update();
-        qApp->processEvents(); // Try to ensure we are on top of any other dialogs
+        pUnzipDialog->repaint(); // Force a redraw
```

Comment:
I'm using a 5MB package and yes that is big enough for the text to flash up (completely filled out) for a short time.  I was vaguely aware that there is the `Qt::WA_OpaquePaintEvent` that is used to tell Qt that a widget repaints it's entirety when redrawn so there is no need to erase it's background when redrawing it but that doesn't seem to be applicable to the `QLabel` involved in the `:/ui/package_manager_unpack.ui` UI file.

I've tried this on my Debian OldStable {NVidia graphics card using the latest Binary Drivers on that platform (3.40) IIRC} in both Debug and Release builds (in case the latter was optimising something that took out a redraw) with a `g++` Version 4.7.2 using the qmake build system for Qt 5.7.0 and also a Debug build on Qt 5.0.2.  I am using `kdm` and `kwin` so I believe I am using a compositing windowing system though I am using `X11` and not `Wayland`.  Is there any obvious difference to your Ubuntu system that might flag up something to cause the difference for you?